lib/lightning_web/live/workflow_live/index.ex

```diff
@@ -62,15 +76,25 @@ defmodule LightningWeb.WorkflowLive.Index do
         socket.assigns.project
       )
 
-    {:ok,
-     socket
-     |> assign(
-       can_delete_workflow: can_delete_workflow,
-       can_create_workflow: can_create_workflow
-     )
-     |> assign_workflow_form(
-       NewWorkflowForm.validate(%{}, socket.assigns.project.id)
-     )}
+    socket
+    |> assign(
+      can_delete_workflow: can_delete_workflow,
+      can_create_workflow: can_create_workflow
+    )
+    |> assign_workflow_form(
+      NewWorkflowForm.validate(%{}, socket.assigns.project.id)
+    )
+    |> then(fn socket ->
+      case RuntimeLimiter.check_limits(%Context{
+             project_id: socket.assigns.project.id
+           }) do
+        :ok ->
+          {:ok, socket}
+
+        {:error, :too_many_runs, %{position: position} = component} ->
```

Comment:
@jyeshe , this strikes me as out of place. if i'm understanding the approach right, we'd apply an `ActionLimiter` here. The user wants to build a new workflow (not run it) and we'd need to check if they are allowed to do so, based on the number of workflows allowed by their plan.

the `too_many_runs` error is what i'd expect to see if someone tried to create a `WorkOrder+Run` or a `Run` for an existing work order.

Reply:
The banner is shown here independently of any action. As a notification/reminder to the user that his plan has reached a limit where `:too_many_runs` is the reason/kind of limit reached.

That's the difference between `check_limits` and `limit_internal` that receives an action as a parameter. I will rename the later to `limit_action`.

Reply:
Done @taylordowns2000, the rename and added a handling for multiple future reasons.